python/lsst/summit/utils/peekExposure.py

```diff
@@ -0,0 +1,772 @@
+    def __init__(self, median):
+        self.median = median
+
+    def getImage(self):
+        return Image(1, 1, self.median, dtype=np.float32)
+
+
+class _MedianSubtractBackgroundConfig(pexConfig.Config):
+    pass
+
+
+class _MedianSubtractBackgroundTask(pipeBase.Task):
+    """Subtract the background from an exposure"""
+
+    ConfigClass = _MedianSubtractBackgroundConfig
+    _DefaultName = "_medianSubtractBackground"
+
+    def run(self, exposure):
+        maskedImage = exposure.getMaskedImage()
+        median = np.nanmedian(maskedImage.image.array)
+        maskedImage.image.array -= median
+
+        return pipeBase.Struct(background=_MedianBackground(median))
+
+
+# Donut PSF; haven't actually tried this yet, but here's a start.
+class DonutPsf(Psf):
+    def __init__(self, size, R_outer, R_inner):
+        Psf.__init__(self, isFixed=True)
+        self.size = size
+        self.R_outer = R_outer
+        self.R_inner = R_inner
+        self.dimensions = Extent2I(size, size)
+
+    def __deepcopy__(self, memo=None):
+        return DonutPsf(self.size, self.R_outer, self.R_inner)
+
+    def resized(self, width, height):
+        assert width == height
+        return DonutPsf(width, self.R_outer, self.R_inner)
+
+    def _doComputeKernelImage(self, position=None, color=None):
+        bbox = self.computeBBox(self.getAveragePosition())
+        img = Image(bbox, type=np.float64)
+        x, y = np.ogrid[bbox.minY : bbox.maxY + 1, bbox.minX : bbox.maxX + 1]
+        rsqr = x**2 + y**2
+        w = (rsqr < self.R_outer**2) & (rsqr > self.R_inner**2)
+        img.array[w] = 1.0
+        img.array /= np.sum(img.array)
+        return img
+
+    def _doComputeBBox(self, position=None, color=None):
+        return Box2I(Point2I(-self.dimensions / 2), self.dimensions)
+
+    def _doComputeShape(self, position=None, color=None):
+        Ixx = self.R_outer**4 - self.R_inner**4
+        Ixx /= self.R_outer**2 - self.R_inner**2
+        return Quadrupole(Ixx, Ixx, 0.0)
+
+    def _doComputeApertureFlux(self, radius, position=None, color=None):
+        return 1 - np.exp(-0.5 * (radius / self.sigma) ** 2)
+
+    def __eq__(self, rhs):
+        if isinstance(rhs, DonutPsf):
+            return (
+                self.size == rhs.size
+                and self.R_outer == rhs.R_outer
+                and self.R_inner == rhs.R_inner
+            )
+        return False
+
+
+class PeekDonutTaskConfig(pexConfig.Config):
+    """Config class for the PeekDonutTask."""
+
+    charImage = pexConfig.ConfigurableField(
+        target=CharacterizeImageTask,
+        doc="CharacterizeImage configuration",
+    )
+    resolution = pexConfig.Field(
+        dtype=float,
+        default=16.0,
+        doc="Target number of pixels for a binned donut",
+    )
+    binSizeMax = pexConfig.Field(
+        dtype=int,
+        default=50,
+        doc="Maximum binning factor for donut mode",
+    )
+    # override installSimplePsf with donut kernel here.
```

Comment:
Is this a TODO entry that was suppose to happen on this branch?

Reply:
More of: started working on it, decided I didn't need it, but didn't want to lose the partial progress.  So my tentative plan was to commit it, and then delete it so it at least lives on in git history.